eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/recommend/EventMeshRecommendImpl.java

```diff
@@ -226,15 +228,16 @@ private String recommendProxy(final Map<String, String> eventMeshMap,
             log.info("eventMeshMap:{},clientDistributionMap:{},group:{}", eventMeshMap, clientDistributionMap, group);
         }
 
-        if (!eventMeshMap.keySet().containsAll(clientDistributionMap.keySet())) {
+        Set<String> clientdDistributionSet=new HashSet<>(clientDistributionMap.keySet());
+        clientdDistributionSet.retainAll(eventMeshMap.keySet());
+        if(!clientdDistributionSet.isEmpty()){
```

Comment:
The retainAll() method of java.util.Set interface is used to retain from this set all of its elements that are contained in the specified collection.
I think the code logic is invalid.

Reply:
Need to invert it to get correct logic when using retainAll()